### PR TITLE
[EASY] [PDF] Define D as a distinct language for the listings package

### PR DESCRIPTION
**Pull request by andralex**
The listings LaTeX package allows code formatting and highlighting in various languages. Previously we hacked the D language keywords on top of listings' C++ definition. This PR defines D as a distinct language.

### CODE REVIEW

**Comment by andralex** [comment]
Is the circleci failure expected?

**Comment by wilzbach** [comment]
> Is the circleci failure expected?

Yes, no one has merged the script yet -> https://github.com/dlang/dlang.org/pull/1948

**Comment by wilzbach** [comment]
FWIW I always use this [lstlangd.tex](https://gist.github.com/wilzbach/19060f7d20a9257676beae35a7481079) file, it's a bit more complete than the one used here.

**Comment by wilzbach** [comment]
Btw it seems like you have included https://github.com/dlang/dlang.org/pull/1951 in this PR too. And we usually use `git rebase` instead of `git merge` to avoid polluting the git history.

**Comment by andralex** [comment]
@wilzbach good pointer. So should I just have issued `git rebase fix-listings`?

**Comment by wilzbach** [comment]
> @wilzbach good pointer. So should I just have issued git rebase fix-listings?

I just tried to improve our contribution guides, hence I am linking to https://wiki.dlang.org/Starting_as_a_Contributor#Working_on_an_pull_request

tl;dr: I prefer keeping a separate upstream remote node:

```
git fetch upstream
git rebase upstream/master
``

**Comment by andralex** [comment]
>FWIW I always use this lstlangd.tex file, it's a bit more complete than the one used here.

Feel free to integrate it later. Thanks!

**Comment by andralex** [comment]
@CyberShadow DAutoTest is failing with:

    Package keyval Error: breakatwhitespace undefined.

I've googled around a bit and found [this](https://tex.stackexchange.com/questions/390599/breakatwhitespace-undefined) discussion, which suggests running:

    sudo apt-get remove fp-utils-3.0.0

Indeed it seems from the logs that the listings-related files are loaded from `/usr/share/texmf/tex/latex/fpc-3.0.0/`, suggesting fp as the culprit.

**Comment by CyberShadow** [comment]
> which suggests running:
> 
>     sudo apt-get remove fp-utils-3.0.0

Um, done.

**Comment by andralex** [comment]
@CyberShadow thanks, that worked!
